test/micro/org/openjdk/bench/java/util/ArrayListBulkOpsBenchmark.java

```diff
@@ -0,0 +1,179 @@
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+/**
+ * Benchmark measuring ArrayList addAll() performance.
+ *
+ * Tests the performance of ArrayList.addAll() when copying from another ArrayList.
+ */
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+@Warmup(iterations = 3, time = 1, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 5, time = 1, timeUnit = TimeUnit.SECONDS)
+@Fork(value = 1, jvmArgs = { "-XX:+UseParallelGC", "-Xmx3g" })
+public class ArrayListBulkOpsBenchmark {
+
+    @Param({ "false", "true" })
+    private boolean enablePoisoning;
+
+    private ArrayList<String> arrayListSource0;
+    private ArrayList<String> arrayListSource5;
+    private ArrayList<String> arrayListSource75;
+    private LinkedList<String> linkedListSource0;
+    private LinkedList<String> linkedListSource5;
+    private LinkedList<String> linkedListSource75;
+    private Set<String> singletonSetSource;
+    private HashMap<String, String> hashMapSource;
+    private TreeSet<String> treeSetSource;
+    private WeakHashMap<String, String> weakHashMapSource;
+
+    @Benchmark
+    public ArrayList<String> addAllArrayList0(Blackhole bh) {
+        ArrayList<String> result = new ArrayList<>(0);
+        result.addAll(arrayListSource0);
+        bh.consume(result);
+        return result;
+    }
+
+    @Benchmark
+    public ArrayList<String> addAllArrayList5(Blackhole bh) {
+        ArrayList<String> result = new ArrayList<>(5);
+        result.addAll(arrayListSource5);
+        bh.consume(result);
+        return result;
+    }
+
+    @Benchmark
+    public ArrayList<String> addAllArrayList75(Blackhole bh) {
+        ArrayList<String> result = new ArrayList<>(75);
+        result.addAll(arrayListSource75);
+        bh.consume(result);
+        return result;
+    }
+
+    @Benchmark
+    public ArrayList<String> addAllLinkedList0(Blackhole bh) {
+        ArrayList<String> result = new ArrayList<>(0);
+        result.addAll(linkedListSource0);
+        bh.consume(result);
+        return result;
+    }
+
+    @Benchmark
+    public ArrayList<String> addAllLinkedList5(Blackhole bh) {
+        ArrayList<String> result = new ArrayList<>(5);
+        result.addAll(linkedListSource5);
+        bh.consume(result);
+        return result;
+    }
+
+    @Benchmark
+    public ArrayList<String> addAllLinkedList75(Blackhole bh) {
+        ArrayList<String> result = new ArrayList<>(75);
+        result.addAll(linkedListSource75);
+        bh.consume(result);
```

Comment:
No need for blackholes, just return the result, it has the same effect I believe.